### DISCUSSION TITLE
toSvg() with attributes instead of inline style.

### DISCUSSION THREAD

**Discussion by juliansimonsson** [fabricjs/fabric.js#7307] [Q&A]
I'm using fabric to make designs to a laserengraver. The engraver is not supporting e.g. style="stroke: red". It only supports SVG when svg-elements uses attributes to describe its features instead of inline styleing, like this: stroke="red". Is there a way (without doing string replace) to ensure the output from toSvg does this?

**Comment by asturur** [comment]
sadly we don't support that.
I think is not hard to change on your end.

**Reply by asturur** [reply]
Look into `_createBaseSVGMarkup`. You will see we add there the style attribute and its content.
That part is pretty much stable.
You can think of overriding that function and get the attribute output rather than the style string

**Reply by juliansimonsson** [reply]
Excellent, thank you that helped.
